tex/generic/pgf/frontendlayer/tikz/libraries/tikzlibrarydecorations.code.tex

```diff
@@ -17,8 +17,10 @@
   /pgf/decoration/.unknown/.code=%
     \pgfifdecoration{\pgfkeyscurrentname}{\edef\tikz@decoration@name{\pgfkeyscurrentname}}
     {\pgfifmetadecoration{\pgfkeyscurrentname}{\edef\tikz@decoration@name{\pgfkeyscurrentname}}
-    {\pgfkeys{/errors/unknown
-        key={/pgf/decoration/\pgfkeyscurrentname}{#1}}}},%
+    % fully expand \pgfkeyscurrentname before being passed to
+    % `/errors/unknown key`
+    {\pgfexpl@exp@args@@Ne\pgfkeys{/errors/unknown
+        key={/pgf/decoration/\pgfkeyscurrentname}{\unexpanded{#1}}}}},%
```

Comment:
If we need `\unexpanded` anyway we are better off using the more idiomatic `/.expanded`.
```suggestion
    {\pgfkeys{/errors/unknown
        key/.expanded={/pgf/decoration/\pgfkeyscurrentname}{\unexpanded{#1}}}}},%
```

Reply:
oops, you're right. Should I remove `pgfutil-expl.tex`?

Reply:
Yes, please. We can do that later in a more coordinated fashion.

Reply:
Unfortunatly,
- `\pgfexpl@exp@args@@Ne\pgfkeys{key={... \pgfkeyscurrentname ...}}` and
- `\pgfkeys{key/.expanded={... \pgfkeyscurrentname ...}}`

are different.  In the first case, value of `\pgfkeyscurrentname` before this `\pgfkeys` is used. While in the second case, value of `\pgfkeyscurrentname` is updated to always hold `.expanded`, the name of just parsed handler.

This change caused the PR failed to fix the original issue, see also https://github.com/pgf-tikz/pgf/issues/1082#issuecomment-1524817940.